<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Proteinfunctionprediction by hongyusu</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Proteinfunctionprediction</h1>
      <h2 class="project-tagline">Project about protein function prediction</h2>
      <a href="https://github.com/hongyusu/ProteinFunctionPrediction" class="btn">View on GitHub</a>
      <a href="https://github.com/hongyusu/ProteinFunctionPrediction/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/hongyusu/ProteinFunctionPrediction/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h1>
<a id="table-of-contents" class="anchor" href="#table-of-contents" aria-hidden="true"><span class="octicon octicon-link"></span></a>Table of Contents</h1>

<ul>
<li><a href="#table-of-content">Table of content</a></li>
<li><a href="#introduction">Introduction</a></li>
<li>
<a href="#data">Data</a>

<ul>
<li><a href="#data-from-master-thesis">Data from Master thesis</a></li>
<li><a href="#other-data">Other data</a></li>
</ul>
</li>
<li>
<a href="#preprocessing">Preprocessing</a>

<ul>
<li><a href="#use-data-from-master-thesis">Use the data from Master thesis</a></li>
<li><a href="#generate-protein-data-with-blast-and-interproscan">Generate protein data with BLAST and InterProScan</a></li>
<li><a href="#data-statistics">Data statistics</a></li>
</ul>
</li>
<li>
<a href="#empirical-evaluation">Empirical evaluation</a>

<ul>
<li>
<a href="#support-vector-machines">Support vector machines</a>

<ul>
<li><a href="#setup">Setup</a></li>
<li><a href="#results">Results</a></li>
</ul>
</li>
</ul>
</li>
</ul>

<h1>
<a id="introduction" class="anchor" href="#introduction" aria-hidden="true"><span class="octicon octicon-link"></span></a>Introduction</h1>

<p>For now, the project aims to reliably predict the function of the transporter proteins.
The function is defined as TC code following the hierarchical structure of the transporter protein classification system.
The hierarchical structure of the classification system has five level. We aim to predict the first four level as the fifth level is the very specific classification.
In addition to the transporter proteins documented in TCDB, we also collect proteins from UniProt.
Uniprot proteins have the following features:</p>

<ol>
<li>Gene Ontology</li>
<li>Protein Family</li>
<li>
<em>BLAST</em> score with UniProt</li>
<li>Taxonomy in NCBI</li>
</ol>

<p>For TCDB proteins we extract features of the following two categorises</p>

<ol>
<li>
<em>BLAST</em> score with TCDB</li>
<li>Many InterProScan features</li>
</ol>

<h1>
<a id="data" class="anchor" href="#data" aria-hidden="true"><span class="octicon octicon-link"></span></a>Data</h1>

<h2>
<a id="data-from-master-thesis" class="anchor" href="#data-from-master-thesis" aria-hidden="true"><span class="octicon octicon-link"></span></a>Data from Master thesis</h2>

<p>We obtain some feature representations for proteins based on the <a href="">Master thesis</a>. This covers features in the following categories. </p>

<ol>
<li>
<p>Features from Gene ontology (GO)</p>

<table>
<thead>
<tr>
<th align="right">Type of data</th>
<th align="right">Number of items</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">Protein</td>
<td align="right">101422</td>
</tr>
<tr>
<td align="right">GO: Biological process</td>
<td align="right">12891</td>
</tr>
<tr>
<td align="right">GO: Molecular function</td>
<td align="right">4816</td>
</tr>
<tr>
<td align="right">GO: Cellular component</td>
<td align="right">1670</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>Protein family (Pfam) data</p>

<table>
<thead>
<tr>
<th align="right">Type of data</th>
<th align="right">Number of items</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">Protein</td>
<td align="right">100589</td>
</tr>
<tr>
<td align="right">Pfam feature</td>
<td align="right">7341</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>Blast score with UniProt</p>

<table>
<thead>
<tr>
<th align="right">Type of data</th>
<th align="right">Number of items</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">Protein</td>
<td align="right">56838</td>
</tr>
<tr>
<td align="right">Blast feature</td>
<td align="right">12646</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>Taxonomy data</p>

<table>
<thead>
<tr>
<th align="right">Type of data</th>
<th align="right">Number of items</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">Protein</td>
<td align="right">104116</td>
</tr>
<tr>
<td align="right">Taxonomy feature</td>
<td align="right">3004</td>
</tr>
</tbody>
</table>
</li>
</ol>

<h2>
<a id="other-data" class="anchor" href="#other-data" aria-hidden="true"><span class="octicon octicon-link"></span></a>Other data</h2>

<p>We extract protein classification data from TCDB database.
As the intersection of the protein data listed above and the ones in TCDB is very small we compute protein features via <em>BLAST</em> and InterProScan.</p>

<ol>
<li>
<p>Transport protein classification data are downloaded from <a href="http://www.tcdb.org/public/">TCDB database</a>. 
NOTE: Su, her you should specify which version has been considered (I guess a version after 7 june)</p>

<table>
<thead>
<tr>
<th align="right">Type of data</th>
<th align="right">Number of items</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">Protein</td>
<td align="right">12515</td>
</tr>
<tr>
<td align="right">TCDB annotation</td>
<td align="right">9456</td>
</tr>
</tbody>
</table>

<ol>
<li>
<p>The annotation of TCDB follows a five level classification hierarchy. The number of classes in each level is shown in the following table</p>

<table>
<thead>
<tr>
<th align="right">Level</th>
<th align="right">Number of classes</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">1</td>
<td align="right">7</td>
</tr>
<tr>
<td align="right">2</td>
<td align="right">30</td>
</tr>
<tr>
<td align="right">3</td>
<td align="right">867</td>
</tr>
<tr>
<td align="right">4</td>
<td align="right">2237</td>
</tr>
<tr>
<td align="right">5</td>
<td align="right">9456</td>
</tr>
</tbody>
</table>
</li>
<li><p>Data file for TCDB sequence and classification information is in the file <code>./Data/tcdb.1</code>.</p></li>
</ol>
</li>
<li>
<p><a href="http://hongyusu.github.io/lessons/2015/06/16/ncbi-blast-installation-and-running-in-parallel/"><em>BLAST</em> with TCDB</a></p>

<ol>
<li>Protein sequences are aligned with themselves by running <em>BLAST</em> algorithms.</li>
<li>This procedure will genrate a pairwise similarity matrix.</li>
<li>Instruction for installing and running <em>BLAST</em> can be found from <a href="http://hongyusu.github.io/lessons/2015/06/16/ncbi-blast-installation-and-running-in-parallel/">my blog post</a>. </li>
<li>In particular, after removing some replicated proteins, there are 12515 protein left in TCDB which will be used to build a TCDB <em>BLAST</em> database.</li>
<li>The cleaned TCDB data file is located in <code>./Data/tcdb</code>.</li>
<li>For the <em>BLAST</em> search, we obtain all hits with e-value below 0.01. </li>
<li>We use <em>BLAST</em> score as similary measure between pair of proteins.</li>
<li>
<p>Some statistics about the TCDB <em>BLAST</em> features are shown in the following table</p>

<table>
<thead>
<tr>
<th align="right">Type of Data</th>
<th align="right">Number of items</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">Protein</td>
<td align="right">12515</td>
</tr>
<tr>
<td align="right">TCDB <em>BLAST</em> feature</td>
<td align="right">12515</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>Data file for TCDB <em>BLAST</em> feature is located as <code>./Data/tcdbblast</code>.</p>

<ol>
<li>
<p>The format of this file is</p>

<p><code>qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore</code>.</p>
</li>
<li>
<p>The keywords are explained in the following table</p>

<table>
<thead>
<tr>
<th align="right">Keyword</th>
<th align="right">Representation</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">qseqid</td>
<td align="right">Query Seq-id</td>
</tr>
<tr>
<td align="right">sseqid</td>
<td align="right">Subject Seq-id</td>
</tr>
<tr>
<td align="right">sallacc</td>
<td align="right">All subject accessions</td>
</tr>
<tr>
<td align="right">qstart</td>
<td align="right">Start of alignment in query</td>
</tr>
<tr>
<td align="right">qend</td>
<td align="right">End of alignment in query</td>
</tr>
<tr>
<td align="right">sstart</td>
<td align="right">Start of alignment in subject</td>
</tr>
<tr>
<td align="right">send</td>
<td align="right">End of alignment in subject</td>
</tr>
<tr>
<td align="right">evalue</td>
<td align="right">Expect value</td>
</tr>
<tr>
<td align="right">bitscore</td>
<td align="right">Bit score</td>
</tr>
<tr>
<td align="right">score</td>
<td align="right">Raw score</td>
</tr>
<tr>
<td align="right">length</td>
<td align="right">Alignment length</td>
</tr>
<tr>
<td align="right">pident</td>
<td align="right">Percentage of identical matches</td>
</tr>
<tr>
<td align="right">mismatch</td>
<td align="right">Number of mismatches</td>
</tr>
<tr>
<td align="right">gapopen</td>
<td align="right">Number of gap openings</td>
</tr>
</tbody>
</table>
</li>
</ol>
</li>
</ol>
</li>
<li>
<p><a href="http://hongyusu.github.io/lessons/2015/06/17/extract-protein-features-via-interproscan/">InterProScan</a></p>

<ol>
<li>For all proteins in TCDB, we extract various protein features by running <a href="https://code.google.com/p/interproscan/">InterProScan</a>.</li>
<li>As this procedure takes time, we would like to install and run the InterProScan on local machines other than on webservers.</li>
<li>A brief instruction for installation and running InterProScan is documented in <a href="http://hongyusu.github.io/lessons/2015/06/17/extract-protein-features-via-interproscan/">my blog post</a>.</li>
<li>
<p>In particular, we download the InterProScan package together with various databases. Version information of InterProScan software and databases can be found from the following table</p>

<table>
<thead>
<tr>
<th align="right">Software and databases</th>
<th align="right">Version</th>
<th align="right">Information</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">InterProScan</td>
<td align="right">5.13-52.0</td>
<td align="right">InterProScan package</td>
</tr>
<tr>
<td align="right">ProDom</td>
<td align="right">2006.1</td>
<td align="right">ProDom is a comprehensive set of protein domain families automatically generated from the UniProt Knowledge Database.</td>
</tr>
<tr>
<td align="right">HAMAP</td>
<td align="right"></td>
<td align="right">High-quality Automated and Manual Annotation of Microbial Proteomes</td>
</tr>
<tr>
<td align="right">SMART</td>
<td align="right">6.2</td>
<td align="right">SMART allows the identification and analysis of domain architectures based on Hidden Markov Models or HMMs</td>
</tr>
<tr>
<td align="right">SuperFamily</td>
<td align="right">1.75</td>
<td align="right">SUPERFAMILY is a database of structural and functional annotation for all proteins and genomes.</td>
</tr>
<tr>
<td align="right">PRINTS</td>
<td align="right">42.0</td>
<td align="right">A fingerprint is a group of conserved motifs used to characterise a protein family</td>
</tr>
<tr>
<td align="right">Panther</td>
<td align="right">9.0</td>
<td align="right">The PANTHER (Protein ANalysis THrough Evolutionary Relationships) Classification System is a unique resource that classifies genes by their functions, using published scientific experimental evidence and evolutionary relationships to predict function even in the absence of direct experimental evidence.</td>
</tr>
<tr>
<td align="right">Gene3d</td>
<td align="right">3.5.0</td>
<td align="right">Structural assignment for whole genes and genomes using the CATH domain structure database</td>
</tr>
<tr>
<td align="right">PIRSF</td>
<td align="right">3.01</td>
<td align="right">The PIRSF concept is being used as a guiding principle to provide comprehensive and non-overlapping clustering of UniProtKB sequences into a hierarchical order to reflect their evolutionary relationships.</td>
</tr>
<tr>
<td align="right">PfamA</td>
<td align="right">27.0</td>
<td align="right">A large collection of protein families, each represented by multiple sequence alignments and hidden Markov models (HMMs)</td>
</tr>
<tr>
<td align="right">PrositeProfiles</td>
<td align="right"></td>
<td align="right">PROSITE consists of documentation entries describing protein domains, families and functional sites as well as associated patterns and profiles to identify them</td>
</tr>
<tr>
<td align="right">TIGRFAM</td>
<td align="right">15.0</td>
<td align="right">TIGRFAMs are protein families based on Hidden Markov Models or HMMs</td>
</tr>
<tr>
<td align="right">PrositePatterns</td>
<td align="right"></td>
<td align="right">PROSITE consists of documentation entries describing protein domains, families and functional sites as well as associated patterns and profiles to identify them</td>
</tr>
<tr>
<td align="right">Coils</td>
<td align="right">2.2</td>
<td align="right">Prediction of Coiled Coil Regions in Proteins</td>
</tr>
<tr>
<td align="right">TMHMM</td>
<td align="right">2.0</td>
<td align="right">Prediction of transmembrane helices in proteins</td>
</tr>
<tr>
<td align="right">Phobius</td>
<td align="right">1.01</td>
<td align="right">A combined transmembrane topology and signal peptide predictor</td>
</tr>
<tr>
<td align="right">SignalP_GRAM_NEGATIVE</td>
<td align="right">4.0</td>
<td align="right">SignalP (organism type gram-negative prokaryotes) predicts the presence and location of signal peptide cleavage sites in amino acid sequences for gram-negative prokaryotes</td>
</tr>
<tr>
<td align="right">SignalP_EUK</td>
<td align="right">4.0</td>
<td align="right">SignalP (organism type eukaryotes) predicts the presence and location of signal peptide cleavage sites in amino acid sequences for eukaryotes.</td>
</tr>
<tr>
<td align="right">SignalP_GRAM_POSITIVE</td>
<td align="right">4.0</td>
<td align="right">SignalP (organism type gram-positive prokaryotes) predicts the presence and location of signal peptide cleavage sites in amino acid sequences for gram-positive prokaryotes</td>
</tr>
</tbody>
</table>
</li>
<li><p>Note that the last five tools and Panthon database are installed into InterProScan manually.</p></li>
<li>It is not necessary to perform again the scanning with InterProScan for all TCDB sequences as most of the TCDB sequences already have UniProt accession number. Therefore, we depend on the lookup service provided by InterProScan in order to directly extract the sequence features from the database.</li>
<li>In addition to direct extraction, protein sequences that is not known to InterProScan are scanned.</li>
<li>
<p>Interproscan results is located as <code>./Data/tcdbips</code>. The file follows the structure described in the following table</p>

<table>
<thead>
<tr>
<th align="right">Column</th>
<th align="right">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">1</td>
<td align="right">Protein Accession (e.g. P51587)</td>
</tr>
<tr>
<td align="right">2</td>
<td align="right">Sequence MD5 digest (e.g. 14086411a2cdf1c4cba63020e1622579)</td>
</tr>
<tr>
<td align="right">3</td>
<td align="right">Sequence Length (e.g. 3418)</td>
</tr>
<tr>
<td align="right">4</td>
<td align="right">Analysis (e.g. Pfam / PRINTS / Gene3D)</td>
</tr>
<tr>
<td align="right">5</td>
<td align="right">Signature Accession (e.g. PF09103 / G3DSA:2.40.50.140)</td>
</tr>
<tr>
<td align="right">6</td>
<td align="right">Signature Description (e.g. BRCA2 repeat profile)</td>
</tr>
<tr>
<td align="right">7</td>
<td align="right">Start location</td>
</tr>
<tr>
<td align="right">8</td>
<td align="right">Stop location</td>
</tr>
<tr>
<td align="right">9</td>
<td align="right">Score - is the e-value of the match reported by member database method (e.g. 3.1E-52)</td>
</tr>
<tr>
<td align="right">10</td>
<td align="right">Status - is the status of the match (T: true)</td>
</tr>
<tr>
<td align="right">11</td>
<td align="right">Date - is the date of the run</td>
</tr>
<tr>
<td align="right">12</td>
<td align="right">InterPro annotations - accession (e.g. IPR002093)</td>
</tr>
<tr>
<td align="right">13</td>
<td align="right">InterPro annotations - description (e.g. BRCA2 repeat)</td>
</tr>
<tr>
<td align="right">14</td>
<td align="right">GO annotations (e.g. GO:0005515)</td>
</tr>
<tr>
<td align="right">15</td>
<td align="right">Pathways annotations (e.g. REACT_71)</td>
</tr>
</tbody>
</table>
</li>
</ol>
</li>
<li><p>Besides the data files described above, all original data are located in the directory <code>./Data/</code></p></li>
</ol>

<h1>
<a id="preprocessing" class="anchor" href="#preprocessing" aria-hidden="true"><span class="octicon octicon-link"></span></a>Preprocessing</h1>

<h2>
<a id="use-the-data-from-master-thesis" class="anchor" href="#use-the-data-from-master-thesis" aria-hidden="true"><span class="octicon octicon-link"></span></a>Use the data from Master thesis</h2>

<ol>
<li>The first strategy we used is to merge the protein data from the Master thesis and from TCDB database to find an intersection of proteins with both various features and classification information in TCDB.</li>
<li>However, we found out that the intersection is very small. Only about 1000 proteins from TCDB has feature representations in the Master thesis data.</li>
<li>
<p>Anywauy, this strategy is briefly illustrated as followings:</p>

<ol>
<li>
<p>Merge datasets from different sources:</p>

<ol>
<li>
<p>Merge the following feature types into one matrix</p>

<table>
<thead>
<tr>
<th align="right">Data type</th>
<th align="right">#Proteins</th>
<th align="right">#Features</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">matgoBP</td>
<td align="right">101422</td>
<td align="right">12891</td>
</tr>
<tr>
<td align="right">matgoCC</td>
<td align="right">101422</td>
<td align="right">1670</td>
</tr>
<tr>
<td align="right">matgoMF</td>
<td align="right">101422</td>
<td align="right">4816</td>
</tr>
<tr>
<td align="right">matblastcompressed</td>
<td align="right">56838</td>
<td align="right">12646</td>
</tr>
<tr>
<td align="right">matpfam</td>
<td align="right">100589</td>
<td align="right">7341</td>
</tr>
<tr>
<td align="right">mattaxo</td>
<td align="right">104116</td>
<td align="right">3004</td>
</tr>
<tr>
<td align="right">TCDB</td>
<td align="right">12516</td>
<td align="right">9456</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>A big global matrix is computed by concatenating the matrices of protein-GO, protein-Blast, protein-Ffam, protein-taxonomy, and protein-TCDB, and protein-TCDBblast.</p>

<ol>
<li><p>The number of proteins and the number of features in the union of the collection of matrices are shown in the following table.</p></li>
<li>
<p>The number of proteins and the number of features in the intersection of the collection of matrics are shown in the following table. Notice that a protein will present in the interection matrix if it has features in GO/<em>BLAST</em>/Pfam/Taxonomy categories. </p>

<table>
<thead>
<tr>
<th align="right">Type</th>
<th align="right">#Proteins</th>
<th align="right">#Features</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">Union</td>
<td align="right">123619</td>
<td align="right">64328</td>
</tr>
<tr>
<td align="right">Intersection</td>
<td align="right">1336</td>
<td align="right">64328</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>Different type of fetures are annotated in <code>.collab</code> according to the following table</p>

<table>
<thead>
<tr>
<th align="right">Feature name</th>
<th align="right">Prefix</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">Gene ontology: biological process</td>
<td align="right">GB</td>
</tr>
<tr>
<td align="right">Gene ontology: cellular component</td>
<td align="right">GC</td>
</tr>
<tr>
<td align="right">Gene ontology: molecular function</td>
<td align="right">GM</td>
</tr>
<tr>
<td align="right">Protein family</td>
<td align="right">PF</td>
</tr>
<tr>
<td align="right"><em>BLAST</em></td>
<td align="right">MB</td>
</tr>
<tr>
<td align="right">Taxonomy</td>
<td align="right">MT</td>
</tr>
<tr>
<td align="right">TCDB classification</td>
<td align="right">TC</td>
</tr>
<tr>
<td align="right">TCDB <em>BLAST</em>
</td>
<td align="right">TB</td>
</tr>
</tbody>
</table>
</li>
</ol>
</li>
</ol>
</li>
</ol>
</li>
</ol>

<h2>
<a id="generate-protein-data-with-blast-and-interproscan" class="anchor" href="#generate-protein-data-with-blast-and-interproscan" aria-hidden="true"><span class="octicon octicon-link"></span></a>Generate protein data with BLAST and InterProScan</h2>

<ol>
<li><p>Realizing that I cannot rely on the data used in the Master thesis, I directly work on proteins from TCDB. In particular, I compuate various protein features via running <em>BLAST</em> search and InterProScan.</p></li>
<li>
<p>First I remove duplicated proteins from the transporter protein classification database (TCDB), particularly, by analyzing the sequence-classification data file.</p>

<p>NOTE: In this way you removed the annotations for proteins having multiple annotation paths. These are 29:</p>

<table>
<thead>
<tr>
<th align="right">AC</th>
<th align="right">TC</th>
</tr>
</thead>
<tbody>
<tr>
<td align="right">D4ZJA6</td>
<td align="right">1.A.30.1.5</td>
</tr>
<tr>
<td align="right">D4ZJA6</td>
<td align="right">1.A.30.1.6</td>
</tr>
<tr>
<td align="right">O24303</td>
<td align="right">1.A.18.1.1</td>
</tr>
<tr>
<td align="right">O24303</td>
<td align="right">3.A.9.1.1</td>
</tr>
<tr>
<td align="right">O34840</td>
<td align="right">2.A.19.2.7</td>
</tr>
<tr>
<td align="right">O34840</td>
<td align="right">2.A.19.2.11</td>
</tr>
<tr>
<td align="right">P0AAW9</td>
<td align="right">8.A.50.1.1</td>
</tr>
<tr>
<td align="right">P0AAW9</td>
<td align="right">2.A.6.2.2</td>
</tr>
<tr>
<td align="right">P0C1S5</td>
<td align="right">9.A.39.1.3</td>
</tr>
<tr>
<td align="right">P0C1S5</td>
<td align="right">9.A.39.1.2</td>
</tr>
<tr>
<td align="right">P18409</td>
<td align="right">1.B.33.3.1</td>
</tr>
<tr>
<td align="right">P18409</td>
<td align="right">1.B.8.6.1</td>
</tr>
<tr>
<td align="right">P23644</td>
<td align="right">3.A.8.1.1</td>
</tr>
<tr>
<td align="right">P23644</td>
<td align="right">1.B.8.2.1</td>
</tr>
<tr>
<td align="right">P25568</td>
<td align="right">2.A.1.24.1</td>
</tr>
<tr>
<td align="right">P25568</td>
<td align="right">9.A.15.1.1</td>
</tr>
<tr>
<td align="right">P28795</td>
<td align="right">9.A.17.1.1</td>
</tr>
<tr>
<td align="right">P28795</td>
<td align="right">3.A.20.1.5</td>
</tr>
<tr>
<td align="right">P29340</td>
<td align="right">3.A.20.1.1</td>
</tr>
<tr>
<td align="right">P29340</td>
<td align="right">3.A.20.1.5</td>
</tr>
<tr>
<td align="right">P34230</td>
<td align="right">3.A.1.203.2</td>
</tr>
<tr>
<td align="right">P34230</td>
<td align="right">3.A.1.203.6</td>
</tr>
<tr>
<td align="right">P94360</td>
<td align="right">3.A.1.1.26</td>
</tr>
<tr>
<td align="right">P94360</td>
<td align="right">3.A.1.1.34</td>
</tr>
<tr>
<td align="right">Q03PY5</td>
<td align="right">3.A.1.28.2</td>
</tr>
<tr>
<td align="right">Q03PY5</td>
<td align="right">3.A.1.26.9</td>
</tr>
<tr>
<td align="right">Q03PY7</td>
<td align="right">3.A.1.28.2</td>
</tr>
<tr>
<td align="right">Q03PY7</td>
<td align="right">3.A.1.26.9</td>
</tr>
<tr>
<td align="right">Q07418</td>
<td align="right">9.A.17.1.1</td>
</tr>
<tr>
<td align="right">Q07418</td>
<td align="right">3.A.20.1.5</td>
</tr>
<tr>
<td align="right">Q15049</td>
<td align="right">9.B.129.1.1</td>
</tr>
<tr>
<td align="right">Q15049</td>
<td align="right">2.A.1.76.1</td>
</tr>
<tr>
<td align="right">Q55JG4</td>
<td align="right">9.B.178.1.5</td>
</tr>
<tr>
<td align="right">Q55JG4</td>
<td align="right">9.B.178.1.6</td>
</tr>
<tr>
<td align="right">Q6GHV7</td>
<td align="right">9.A.39.1.1</td>
</tr>
<tr>
<td align="right">Q6GHV7</td>
<td align="right">9.A.39.1.2</td>
</tr>
<tr>
<td align="right">Q6GUB0</td>
<td align="right">2.A.88.1.1</td>
</tr>
<tr>
<td align="right">Q6GUB0</td>
<td align="right">3.A.1.25.1</td>
</tr>
<tr>
<td align="right">Q72L52</td>
<td align="right">3.A.1.1.24</td>
</tr>
<tr>
<td align="right">Q72L52</td>
<td align="right">3.A.1.1.25</td>
</tr>
<tr>
<td align="right">Q8EAG6</td>
<td align="right">1.A.30.1.5</td>
</tr>
<tr>
<td align="right">Q8EAG6</td>
<td align="right">1.A.30.1.6</td>
</tr>
<tr>
<td align="right">Q8KQR1</td>
<td align="right">9.A.39.1.4</td>
</tr>
<tr>
<td align="right">Q8KQR1</td>
<td align="right">9.A.39.1.2</td>
</tr>
<tr>
<td align="right">Q99257</td>
<td align="right">1.I.1.1.1</td>
</tr>
<tr>
<td align="right">Q99257</td>
<td align="right">3.A.22.1.1</td>
</tr>
<tr>
<td align="right">Q9FBS5</td>
<td align="right">3.A.1.1.42</td>
</tr>
<tr>
<td align="right">Q9FBS5</td>
<td align="right">3.A.1.1.43</td>
</tr>
<tr>
<td align="right">Q9FBS6</td>
<td align="right">3.A.1.1.42</td>
</tr>
<tr>
<td align="right">Q9FBS6</td>
<td align="right">3.A.1.1.43</td>
</tr>
<tr>
<td align="right">Q9FBS7</td>
<td align="right">3.A.1.1.42</td>
</tr>
<tr>
<td align="right">Q9FBS7</td>
<td align="right">3.A.1.1.43</td>
</tr>
<tr>
<td align="right">Q9L0Q1</td>
<td align="right">3.A.1.1.33</td>
</tr>
<tr>
<td align="right">Q9L0Q1</td>
<td align="right">3.A.1.1.36</td>
</tr>
<tr>
<td align="right">Q9RL01</td>
<td align="right">2.A.1.72.1</td>
</tr>
<tr>
<td align="right">Q9RL01</td>
<td align="right">2.A.1.15.15</td>
</tr>
<tr>
<td align="right">Q9Y3Q4</td>
<td align="right">1.A.1.5.10</td>
</tr>
<tr>
<td align="right">Q9Y3Q4</td>
<td align="right">1.A.1.5.11</td>
</tr>
</tbody>
</table>

<p>If we consider only the annotation till to the fourth level we have only 13 proteins with multiple annotated paths, but in this way we miss (considering annotations from the first to the fourth level) 41 annotations. To my opinion it is not a good idea to eliminate such annotations. Considering that in practice there are only a few multiple paths, we can ignore multiple paths in our predictions, but without removing them from the data. You can recover the full TCDB annotations using the file tcdb.1 instead of tcdb in the Data directory.</p>

<p>The TCDB annotation matrix, including all the duplicated path annotations till to the fourth level of the TCDB is available in Preprocessing/Results/tcdb1.annotations.levels1234.txt.gz.
This gzipped text file has Protein AC on rows (12546 AC) and 3145 TCDB classes on columns. This is a 0/1 dense matrix: the entry (i,j) is equal to 1 means that protein i is annotated with TCDB class j, otherwise (i,j) = 0.</p>
</li>
<li><p>Then I run <em>BLAST</em> search and InterProScan for all preprocessed proteins.</p></li>
<li>Merge TCDB classification (protein labels), TCDB <em>BLAST</em> features, and TCDB InterProScan features.</li>
<li>Make data matrices of different types, e.g., different feature matrices and label matrix.

<ol>
<li>In particular, <em>BLAST</em> features use real number (<em>BLAST</em> score), other InterProScan feature use integer number as count</li>
</ol>
</li>
<li>
<p>Important scripts and result files are listed as follows.</p>

<pre><code>Preprocessing
|---Bins
   |---process_tcdb.py        # process original TCDB database (remove duplication ect)
   |---merge_tcdb_blast_and_ips.py          # merge TCDB blast, ips and classfiication data
   |---run_blast.sh           # run _BLAST_ search
   |---run_interproscan.sh    # run interproscan search to generate protein features
   |---separate_different_features.py # generate feature matrices of different types
|---Results
   |---tcdbdata               # merged data in sparse matrix format: 'protein name' 'feature name' 'value' 
   |---tcdbdata.collab        # feature names
   |---tcdbdata.rowlab        # protein names
   |---tcdbdata.mtx           # sparse data matrix with format 'protein id' 'feature id' 'value'
      |---tcdb1.annotations.gz   # dense matrix (text file gzipped) with Swissprot AC on rows and TCDB classes on columns. 
                                 # Entry (i,j) of the matrix is 1 if protein i is annotated to TC class j, otherwise (i,j) = 0
                                 # this matrix includes all the annotations (including multiple paths) obtained from the file tcdb.1 in the Data directory.
Experiments
|---Data
   |---tcdb.prefix       # data matrices of different type, where type information are explained in the section of Data statistics.
   |---README.md         # read me file for experimental data
   |---tcdb.collab       # feature names
   |---tcdb.rowlab       # protein names
|---Data.tar.gz           # compressed Data files, including files in `./Data` folder
</code></pre>
</li>
</ol>

<h2>
<a id="data-statistics" class="anchor" href="#data-statistics" aria-hidden="true"><span class="octicon octicon-link"></span></a>Data statistics</h2>

<ol>
<li>
<p>I compute the following statistice for the overall merged dataset, in particular, for the file <code>./Preprocessing/Results/tcdb.mtx</code>.</p>

<table>
<thead>
<tr>
<th align="left">Type of statistics</th>
<th align="right">Value</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">Number of proteins</td>
<td align="right">12546</td>
</tr>
<tr>
<td align="left">Number of features</td>
<td align="right">25704</td>
</tr>
<tr>
<td align="left">Number of categories</td>
<td align="right">20</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>Category information are listed in the table as follows: </p>

<table>
<thead>
<tr>
<th align="left">Feature prefix</th>
<th align="right">Size</th>
<th align="center">Feature type</th>
<th align="center">Version</th>
<th align="left">Feature description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">TC__</td>
<td align="right">3145</td>
<td align="center">TCDB</td>
<td align="center"></td>
<td align="left">TCDB classification</td>
</tr>
<tr>
<td align="left">TB__</td>
<td align="right">12535</td>
<td align="center"><em>BLAST</em></td>
<td align="center"></td>
<td align="left">
<em>BLAST</em> search</td>
</tr>
<tr>
<td align="left">TIProDom__</td>
<td align="right">145</td>
<td align="center">ProDom</td>
<td align="center">2006.1</td>
<td align="left">ProDom is a comprehensive set of protein domain families automatically generated from the UniProt Knowledge Database.</td>
</tr>
<tr>
<td align="left">TIHamap__</td>
<td align="right">209</td>
<td align="center">HAMAP</td>
<td align="center"></td>
<td align="left">High-quality Automated and Manual Annotation of Microbial Proteomes</td>
</tr>
<tr>
<td align="left">TISMART__</td>
<td align="right">240</td>
<td align="center">SMART</td>
<td align="center">6.2</td>
<td align="left">SMART allows the identification and analysis of domain architectures based on Hidden Markov Models or HMMs</td>
</tr>
<tr>
<td align="left">TISUPERFAMILY__</td>
<td align="right">512</td>
<td align="center">SuperFamily</td>
<td align="center">1.75</td>
<td align="left">SUPERFAMILY is a database of structural and functional annotation for all proteins and genomes.</td>
</tr>
<tr>
<td align="left">TIPRINTS__</td>
<td align="right">579</td>
<td align="center">PRINTS</td>
<td align="center">42.0</td>
<td align="left">A fingerprint is a group of conserved motifs used to characterise a protein family</td>
</tr>
<tr>
<td align="left">TIPANTHER__</td>
<td align="right">4070</td>
<td align="center">Panther</td>
<td align="center">9.0</td>
<td align="left">The PANTHER (Protein ANalysis THrough Evolutionary Relationships) Classification System is a unique resource that classifies genes by their functions, using published scientific experimental evidence and evolutionary relationships to predict function even in the absence of direct experimental evidence.</td>
</tr>
<tr>
<td align="left">TIGene3D__</td>
<td align="right">611</td>
<td align="center">Gene3d</td>
<td align="center">3.5.0</td>
<td align="left">Structural assignment for whole genes and genomes using the CATH domain structure database</td>
</tr>
<tr>
<td align="left">TIPIRSF__</td>
<td align="right">283</td>
<td align="center">PIRSF</td>
<td align="center">3.01</td>
<td align="left">The PIRSF concept is being used as a guiding principle to provide comprehensive and non-overlapping clustering of UniProtKB sequences into a hierarchical order to reflect their evolutionary relationships.</td>
</tr>
<tr>
<td align="left">TIPfam__</td>
<td align="right">2025</td>
<td align="center">PfamA</td>
<td align="center">27.0</td>
<td align="left">A large collection of protein families, each represented by multiple sequence alignments and hidden Markov models (HMMs)</td>
</tr>
<tr>
<td align="left">TIProSiteProfiles__</td>
<td align="right">282</td>
<td align="center">PrositeProfiles</td>
<td align="center"></td>
<td align="left">PROSITE consists of documentation entries describing protein domains, families and functional sites as well as associated patterns and profiles to identify them</td>
</tr>
<tr>
<td align="left">TITIGRFAM__</td>
<td align="right">769</td>
<td align="center">TIGRFAM</td>
<td align="center">15.0</td>
<td align="left">TIGRFAMs are protein families based on Hidden Markov Models or HMMs</td>
</tr>
<tr>
<td align="left">TIProSitePatterns__</td>
<td align="right">285</td>
<td align="center">PrositePatterns</td>
<td align="center"></td>
<td align="left">PROSITE consists of documentation entries describing protein domains, families and functional sites as well as associated patterns and profiles to identify them</td>
</tr>
<tr>
<td align="left">TICoils__</td>
<td align="right">1</td>
<td align="center">Coils</td>
<td align="center">2.2</td>
<td align="left">Prediction of Coiled Coil Regions in Proteins</td>
</tr>
<tr>
<td align="left">TITMHMM__</td>
<td align="right">1</td>
<td align="center">TMHMM</td>
<td align="center">2.0</td>
<td align="left">Prediction of transmembrane helices in proteins</td>
</tr>
<tr>
<td align="left">TIPhobius__</td>
<td align="right">7</td>
<td align="center">Phobius</td>
<td align="center">1.01</td>
<td align="left">A combined transmembrane topology and signal peptide predictor</td>
</tr>
<tr>
<td align="left">TISignalP_GRAM_NEGATIVE__</td>
<td align="right">2</td>
<td align="center">SignalP_GRAM_NEGATIVE</td>
<td align="center">4.0</td>
<td align="left">SignalP (organism type gram-negative prokaryotes) predicts the presence and location of signal peptide cleavage sites in amino acid sequences for gram-negative prokaryotes</td>
</tr>
<tr>
<td align="left">TISignalP_EUK__</td>
<td align="right">2</td>
<td align="center">SignalP_EyUK</td>
<td align="center">4.0</td>
<td align="left">SignalP (organism type eukaryotes) predicts the presence and location of signal peptide cleavage sites in amino acid sequences for eukaryotes.</td>
</tr>
<tr>
<td align="left">TISignalP_GRAM_POSITIVE__</td>
<td align="right">1</td>
<td align="center">SignalP_GRAM_POSITIVE</td>
<td align="center">4.0</td>
<td align="left">SignalP (organism type gram-positive prokaryotes) predicts the presence and location of signal peptide cleavage sites in amino acid sequences for gram-positive prokaryotes</td>
</tr>
</tbody>
</table>
</li>
</ol>

<h1>
<a id="empirical-evaluation" class="anchor" href="#empirical-evaluation" aria-hidden="true"><span class="octicon octicon-link"></span></a>Empirical evaluation</h1>

<h2>
<a id="support-vector-machines" class="anchor" href="#support-vector-machines" aria-hidden="true"><span class="octicon octicon-link"></span></a>Support vector machines</h2>

<h2>
<a id="setup" class="anchor" href="#setup" aria-hidden="true"><span class="octicon octicon-link"></span></a>Setup</h2>

<ol>
<li>Linear SVM is used as the baseline learner. In particular, we use SVM implementation from <a href="https://www.csie.ntu.edu.tw/%7Ecjlin/libsvm/">libSVM</a>.</li>
<li>We select for each input feature a SVM margin slack (C) parameter from the range {0.01,0.1,1,10,100}.</li>
<li>Parameter selection procedure is baed on 5000 randomly selected proteins from the original dataset.</li>
<li>After parametere selection, the best SVM C parameter is applied to train a SVM classifier.</li>
<li>Experimental results are reported from a five fold cross validation procedure. We randomly divide examples into five disjoin set of equal size. In each iteration, we use one set for testing and the rest for training. The same procedure is then repeated five times.</li>
<li>We report the following metrics to measure the performance of the classifier including AUC, accuracy, F1, precision, and recall. The metrics are computed based on microlabels.</li>
</ol>

<h2>
<a id="results" class="anchor" href="#results" aria-hidden="true"><span class="octicon octicon-link"></span></a>Results</h2>

<ol>
<li>
<p>Preliminary experimental results are shown in the following table</p>

<table>
<thead>
<tr>
<th align="center">Input</th>
<th align="center">AUC</th>
<th align="center">Accuracy</th>
<th align="center">F1</th>
<th align="center">Precision</th>
<th align="center">Recall</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">../Data/tcdb.TB</td>
<td align="center">0.9641</td>
<td align="center">0.9992</td>
<td align="center">0.7445</td>
<td align="center">0.9786</td>
<td align="center">0.6008</td>
</tr>
<tr>
<td align="center">../Data/tcdb.TICoils</td>
<td align="center">0.8541</td>
<td align="center">0.9964</td>
<td align="center">0.0356</td>
<td align="center">0.0380</td>
<td align="center">0.0336</td>
</tr>
<tr>
<td align="center">../Data/tcdb.TIGene3D</td>
<td align="center">0.8815</td>
<td align="center">0.9881</td>
<td align="center">0.0403</td>
<td align="center">0.0240</td>
<td align="center">0.1266</td>
</tr>
<tr>
<td align="center">../Data/tcdb.TIHamap</td>
<td align="center">0.8691</td>
<td align="center">0.9981</td>
<td align="center">0.0553</td>
<td align="center">0.9498</td>
<td align="center">0.0285</td>
</tr>
<tr>
<td align="center">../Data/tcdb.TIPANTHER</td>
<td align="center">0.9081</td>
<td align="center">0.9981</td>
<td align="center">0.4240</td>
<td align="center">0.5239</td>
<td align="center">0.3562</td>
</tr>
<tr>
<td align="center">../Data/tcdb.TIPfam</td>
<td align="center">0.9415</td>
<td align="center">0.9978</td>
<td align="center">0.3822</td>
<td align="center">0.4201</td>
<td align="center">0.3506</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>AUC curve is shown as </p>

<p><img src="https://github.com/aalto-ics-kepaco/ProteinFunctionPrediction/blob/master/Experiments/Plots/auc.jpg" alt="alt text"></p>
</li>
</ol>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/hongyusu/ProteinFunctionPrediction">Proteinfunctionprediction</a> is maintained by <a href="https://github.com/hongyusu">hongyusu</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

            <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-64736416-1");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>

  </body>
</html>
